src/citrine/ara/variables.py

```diff
@@ -0,0 +1,98 @@
+"""Variable definitions for Ara."""
+from typing import Type, Optional, List  # noqa: F401
+from abc import abstractmethod
+
+from taurus.entity.link_by_uid import LinkByUID
+
+from citrine._serialization.serializable import Serializable
+from citrine._serialization.polymorphic_serializable import PolymorphicSerializable
+from citrine._serialization import properties
+
+
+class Variable(PolymorphicSerializable['Variable']):
+    """A variable that can be assigned values present in material histories.
+
+    Abstract type that returns the proper type given a serialized dict.
+    """
+
+    @abstractmethod
+    def _attrs(self) -> List[str]:
+        pass  # pragma: no cover
+
+    def __eq__(self, other):
+        try:
+            return all([
+                self.__getattribute__(key) == other.__getattribute__(key) for key in self._attrs()
+            ])
```

Comment:
If `other` has matches all of the attributes that `self` has, but has other attributes in addition, then I believe this will erroneously return True.

Reply:
That's why `type` is included to make sure that the classes being compared are intended to be equivalent.  Note that I lifted this code block from [orion descriptors](https://github.com/CitrineInformatics/citrine-python/blob/develop/src/citrine/informatics/descriptors.py#L46).